From d61aecc18bde28477280ee7ff6cd408ac069cc53 Mon Sep 17 00:00:00 2001
From: Markus Lehtonen <markus.lehtonen@linux.intel.com>
Date: Mon, 16 Oct 2017 14:17:04 +0300
Subject: [PATCH] specify install rpath via environment

If defined, use the value of environment variable MESON_INSTALL_RPATH as
'install_rpath' configuration setting.

Signed-off-by: Markus Lehtonen <markus.lehtonen@linux.intel.com>
---
 mesonbuild/build.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/build.py b/mesonbuild/build.py
index a5ebc343..296e2ccd 100644
--- a/mesonbuild/build.py
+++ b/mesonbuild/build.py
@@ -659,7 +659,7 @@ class BuildTarget(Target):
             if not(os.path.isfile(trial)):
                 raise InvalidArguments('Tried to add non-existing extra file %s.' % i)
         self.extra_files = extra_files
-        self.install_rpath = kwargs.get('install_rpath', '')
+        self.install_rpath = os.environ.get("MESON_INSTALL_RPATH") or kwargs.get('install_rpath', '')
         if not isinstance(self.install_rpath, str):
             raise InvalidArguments('Install_rpath is not a string.')
         resources = kwargs.get('resources', [])
-- 
2.13.6

